pkgs/top-level/lua-packages.nix

```diff
@@ -33,12 +33,19 @@ let
       sha256 = "16fffbrgfcw40kskh2bn9q7m3gajffwd2f35rafynlnd7llwj1qj";
     };
 
+    patchPhase = ''
+      sed -e "s,MACOSX_DEPLOYMENT_TARGET=10.4,MACOSX_DEPLOYMENT_TARGET=10.10," -i Makefile
```

Comment:
In general, it would be nice to not specify this on a per-package basis. Does it pick it up from the environment if you just tell the makefile not to set it? the stdenv specifies its own value for `MACOSX_DEPLOYMENT_TARGET`. Doing it that way gives us an easier way to declare what Mac OS is supported and control all of nixpkgs as a whole.